asf_heat_pump_suitability/analysis/hn_zones/comparison_of_hn_zones.py

```diff
@@ -0,0 +1,350 @@
+) -> Tuple[pl.DataFrame, List[str]]:
+    """
+    Load and process Nesta heat pump suitability data for a specific local authority.
+
+    Args:
+        nesta_hp_suitability_scores (str): Path to the Nesta heat pump suitability Parquet file with HN scores.
+        local_authority (str): Local authority name to filter the data.
+
+    Returns:
+        Tuple[pl.DataFrame, List[str]]:
+            - DataFrame containing filtered LSOA codes and 'HN_N_avg_score_weighted' column.
+            - List of unique LSOA codes within the local authority.
+    """
+    # Load the data from the Parquet file
+    hp_scores_df = pl.read_parquet(nesta_hp_suitability_scores)
+
+    # Rename 'lsoa' to 'LSOA21CD' if needed to match the GeoDataFrame
+    if "lsoa" in hp_scores_df.columns:
+        hp_scores_df = hp_scores_df.rename({"lsoa": "LSOA21CD"})
+
+    # Filter by local authority and select relevant columns
+    la_filtered_scores = hp_scores_df.filter(
+        pl.col("lsoa_name").str.starts_with(local_authority)
+    ).select(["LSOA21CD", "HN_N_avg_score_weighted"])
+
+    # Extract unique LSOA codes
+    la_unique_lsoas = la_filtered_scores["LSOA21CD"].unique().to_list()
+
+    return la_filtered_scores, la_unique_lsoas
+
+
+def add_desnz_pilot_score(
+    la_hp_suitability_scores: pl.DataFrame, list_of_desnz_hn_lsoas: List[str]
+) -> Tuple[pl.DataFrame, float, float]:
+    """
+    Add a 'DESNZ_pilot_score' column to the Nesta heat pump suitability data and compute average scores.
+
+    Args:
+        la_hp_suitability_scores (pl.DataFrame): Data containing 'LSOA21CD' and 'HN_N_avg_score_weighted' columns.
+        list_of_desnz_hn_lsoas (List[str]): List of LSOA codes within the DESNZ HN pilot zones.
+
+    Returns:
+        Tuple[pl.DataFrame, float, float]:
+            - Updated DataFrame with 'DESNZ_pilot_score' column added (1 if in pilot zone, otherwise 0).
+            - Average 'HN_N_avg_score_weighted' for rows where 'DESNZ_pilot_score' is 1.
+            - Average 'HN_N_avg_score_weighted' for rows where 'DESNZ_pilot_score' is 0.
+    """
+    # Add 'DESNZ_pilot_score' column: 1 if LSOA is in DESNZ pilot zone, else 0
+    la_hp_suitability_scores = la_hp_suitability_scores.with_columns(
+        pl.col("LSOA21CD")
+        .is_in(list_of_desnz_hn_lsoas)
+        .cast(pl.Int8)
+        .alias("DESNZ_pilot_score")
+    )
+    # Calculate average scores where 'DESNZ_pilot_score' == 1 and == 0
+    avg_hn_score_pilot_1 = _calculate_hn_pilot_average_score(
+        la_hp_suitability_scores, pilot_score=1
+    )
+    avg_hn_score_pilot_0 = _calculate_hn_pilot_average_score(
+        la_hp_suitability_scores, pilot_score=0
+    )
+    return la_hp_suitability_scores, avg_hn_score_pilot_1, avg_hn_score_pilot_0
+
+
+def _calculate_hn_pilot_average_score(
+    la_hp_suitability_scores: pl.DataFrame, pilot_score: int
+) -> float:
+    """
+    Calculate the average 'HN_N_avg_score_weighted' for a specified 'DESNZ_pilot_score' value.
```

Comment:
I would maybe reword this to make it a bit clearer. Something like:
```suggestion
    Calculate the average Nesta Heat Network score for LSOAs in (`DESNZ_pilot_score == 1`) or not in  (`DESNZ_pilot_score == 0`) DESNZ heat network pilot areas. 
```

Reply:
Thanks for this, great suggestion for the doc string!